doc/lsst.obs.base/creating-an-obs-package.rst

```diff
@@ -0,0 +1,188 @@
+
+Create a file, ``python/lsst/obs/example/rawFormatter.py``, containing an ``ExampleRawFormatter`` derived from the `FitsRawFormatterBase` base class.
+At minimum, you must define a ``translatorClass`` pointing to the ``ExampleTranslator`` you made in :ref:`translator` and a ``filterDefinitions`` pointing to the filter list you created in :ref:`filters`.
+This formatter can also contain specializations for your specific camera, for example to rotate the `~lsst.afw.geom.SkyWcs` for detectors in the camera that have a different orientation.
+
+.. code-block:: python
+
+    __all__ = ["ExampleCameraRawFormatter"]
+
+    from astro_metadata_translator import ExampleTranslator
+    from lsst.obs.base.fitsRawFormatterBase import FitsRawFormatterBase
+    from . import EXAMPLE_FILTER_DEFINITIONS
+
+
+    class ExampleCameraRawFormatter(FitsRawFormatterBase):
+        translatorClass = ExampleTranslator
+        filterDefinitions = EXAMPLE_FILTER_DEFINITIONS
+
+        def getDetector(self, id):
+            return ExampleCamera().getCamera()[id]
+
+.. _instrument:
+
+Instrument
+==========
+
+An `Instrument` defines the instrument-specific logic for the Data Butler.
+
+First create a new file ``tests/test_instrument.py`` with a test case derived from `~lsst.obs.base.instrument_tests.InstrumentTests` and `~lsst.utils.tests.TestCase`, defining ``self.data`` and ``self.instrument`` in ``setUp``.
+The `set` of ``physical_filters`` you provide here will be checked to ensure that your `FilterDefinitionCollection` is loaded correctly.
+
+.. code-block:: python
+
+    """Tests of the ExampleCam instrument class.
+    """
+
+    import unittest
+
+    import lsst.utils.tests
+    import lsst.obs.example
+    from lsst.obs.base.instrument_tests import InstrumentTests, InstrumentTestData
+
+
+    class TestExampleCam(InstrumentTests, lsst.utils.tests.TestCase):
+        def setUp(self):
+            physical_filters = {"example g filter",
+                                "example z filter"}
+
+            self.data = InstrumentTestData(name="Example",
+                                           nDetectors=4,
+                                           firstDetectorName="1_1",
+                                           physical_filters=physical_filters)
+            self.instrument = lsst.obs.example.ExampleCam()
+
+
+    def setup_module(module):
+        lsst.utils.tests.init()
+
+    if __name__ == '__main__':
+        lsst.utils.tests.init()
+        unittest.main()
+
+Run this test via
+
+.. code-block:: bash
+
+    pytest -sv tests/test_instrument.py
+
+the tests should fail, as there is no Example `Instrument` yet.
+
+Next, add a file in ``python/lsst/obs/example/instrument.py`` containing a subclass of `Instrument`, `ExampleCam`, which at minimum overrides these abstract methods: `Instrument.getName`, `Instrument.getCamera`, `Instrument.register`, `Instrument.filterDefinitions`, and define ``self.configPaths`` in ``__init__``.
```

Comment:
Is the `ExampleCam` here the name of the `Instrument` subclass? I'd expect that to be `ExampleCamInstrument` (and double backticks).

Reply:
For HSC and DECam, we named them "HyperSuprimeCam" and "DarkEnergyCamera". I see you did `LsstCamInstrument` for LSST. Should we change the former two?

Reply:
I think I would understand it better if the parenthetical said ", named ``ExampleCam``,"

Reply:
Do we want to go with `ExampleCamInstrument` here and update HSC/DECam? Now's our chance to enshrine the "correct" way for the future. I don't have a strong opinion one way or the other.